thrall/app/lib/ThrallMessageConsumer.scala

```diff
@@ -83,9 +94,62 @@ class ThrallMessageConsumer(config: ThrallConfig, es: ElasticSearch, thrallMetri
   def deleteAllUsages(usage: JsValue) =
     Future.sequence( withImageId(usage)(id => es.deleteAllImageUsages(id)) )
 
-  def updateRcsRights(rights: JsValue) =
-    Future.sequence( withImageId(rights)(id => es.updateImageSyndicationRights(id, rights \ "data")) )
+  def refreshInferredSyndicationRights(imageJson: JsValue) = {
+    Future.sequence(
+      withImageId(imageJson) { id =>
+        val image = imageJson.as[Image]
+        GridLogger.info(s"upserting inferred rights", id)
+        es.updateImageSyndicationRights(id, image.syndicationRights)
+      }
+    )
+  }
+
+  def upsertSyndicationRights(rights: JsValue) = {
+    (rights \ "data").validate[SyndicationRights].fold(
+      err => {
+        val msg = s"Unable to parse message as SyndicationRights ${JsError.toJson(err).toString}"
+        GridLogger.error(msg)
+        Future.failed(SNSBodyParseError(msg))
+      },
+      rightsFromRcs => withImageId(rights) { id =>
+        es.getImage(id) map {
+          case Some(image) => {
+            GridLogger.info(s"Upserting syndication rights from RCS feed", image.id)
+            es.updateImageSyndicationRights(id, Some(rightsFromRcs))
+
+            image.userMetadata.map{edits => {
+              edits.photoshoot match {
+                case Some(photoshoot) if !rightsFromRcs.isInferred => {
```

Comment:
Maybe the naming is not great here as while reading it I was expecting `rightsFromRcs` to never be inferred. Maybe `syndicationRights` works better in this case?